### DISCUSSION TITLE
Recommended way to access cross account access to AWS resources Java SDK

### DISCUSSION THREAD

**Discussion by Shlok-Chaurasia** [aws/aws-sdk-java#2824] [Q&A] [answered]
Have a usecase where there is need to access cross account DynamoDB, S3 access in Account A from lambda/ECS in Account B. With AWS Java SDK2 what is the recommended Credential Provider to use for this use case?
There are multiple options:
https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html

1. STSAssumeRoleSessionCredentialsProvider
2. default credential provider chain 
etc.

**Comment by debora-ito** [comment, marked as answer]
Hi @Shlok-Chaurasia apologies for the delay in response here.

The recommended way is to use STSAssumeRoleSessionCredentialsProvider.

The default credential provider chain won't work in this case, the STSAssumeRole is not in the chain.
